R/select_vpfiles.R

```diff
@@ -1,66 +1,85 @@
-#' Select vertical profiles files (\code{vp}) from computer
+#' Select vertical profile (\code{vp}) files from computer
 #'
-#' Collect a list of vp file names within a directory that comply to the given
-#' country, radar and date range combination
+#' Create a list of vertical profile (\code{vp}) files from a local directory
+#' that match a specific date and radar range. Files are selected based on their
+#' file name (not directory structure), which should be of format
+#' \code{radar_vp_yyyymmdd*.*}, such as \code{bewid_vp_20171123T1900Z_0x5.h5}.
 #'
-#' @param directory Main directory to look into recusively.
-#' @param date_min ISO format date as start of the vp file query.
-#' @param date_max ISO format date as end of the vp file query.
-#' @param country Character vector with two letter country shortcuts.
-#' @param radar Character vector with three letter radar sindicators. This
-#' can be defined independently from the countries named.
+#' @param date_min character. YYYY-MM-DD start date of file selection.
+#' @param date_max character. YYYY-MM-DD end date of file selection.
+#' @param radars character (vector). 5-letter country/radar code(s) (e.g.
+#'   \code{bejab}) of radars to include in file selection.
+#' @param directory character. Path to local directory where files should be
+#'   looked for.
 #'
-#' @return Character list of filenames that comply to the given
-#' radar/country and date range query
+#' @return Character vector of file paths that comply to the given date and
+#'   radar range.
 #'
 #' @export
-#' @importFrom lubridate as_date
 #'
 #' @examples
-#' my_directory <- "~/my/directory/"
-#' select_vpfiles(my_directory, "2016-10-01", "2017-01-31", c("be"))
-select_vpfiles <- function(directory, date_min, date_max, country = NULL,
-                           radar = NULL) {
-  if (is.null(country)) {
-    country <- "([a-z]{2})"
+#' my_path <- "~/my/directory"
+#' select_vpfiles(
+#'   date_min = "2016-10-03",
+#'   date_max = "2016-10-05",
+#'   radars = "bejab",
+#'   directory = my_path
+#' )
+select_vpfiles <- function(date_min = NULL, date_max = NULL, radars = NULL,
```

Comment:
When people do not ass the keyword arguments, sequence matters in R. So make sure that the code is backward compatible instead of switching argument position of funcitons in between releases

Reply:
A bit confused by https://github.com/adokter/bioRad/issues/173#issuecomment-445461125 but maybe it' only a Python thing that named arguments should not preceed not-named arguments...

Reply:
We basically want to answer:

1. Do we want to align the order of `select_vpfiles()` and `download_vpfiles()`. My 2 cents: yes.
2. If so, only the position of `directory` (and parameter `overwrite`) is different: do we want it in front or back. My 2 cents: choose order `date_min`, `date_max`, `radars`, `directory`, (`overwrite`) because https://github.com/adokter/bioRad/issues/173#issuecomment-445495121).
3. And for backwards compatibility: several things have changed: `radar` -> `radars`, `country` -> included in `radars`, `directory` -> to end. All these are now [captured in the deprecation](https://github.com/adokter/bioRad/pull/176/files#diff-2a6c884158d6474f545d10290280f1bb) of `retrieve_vp_paths()` and `download_vp()`. Since this change follows fairly quickly after our latest deprecation and any error message is pretty straightforward, I though to bundle these with our latest deprecation.